**Hypothesis testing**

My capstone project is intended to predict prices for 2 different periods (off-peak and peak) for three different seasons (summer, winter, and shoulder). 

For this report I intend to determine whether or not there is a statistically significant difference in the mean price between peak and off peak for each of the three seasons.

My hypothesis is that the prices between peak and off peak are statisticall significant.

My null hypotheisis is that prices are not different between the two periods.

In [53]:
import pandas as pd
import math
import numpy as np
import scipy.stats as st
import statsmodels.api as sm

data= pd.read_csv(r'C:\Users\Pujanande\Desktop\Capstone 1\cap1_training_data.csv') #read in training data

In [54]:
summer = data['FERC_season'] == 'summer' #sample summer
summer_df = data[summer][['FERC_time', 'PRICEINDOLPERMWH']]

winter = data['FERC_season'] == 'winter' #sample winter
winter_df = data[winter][['FERC_time', 'PRICEINDOLPERMWH']]

shoulder = data['FERC_season'] == 'shoulder' #sample shoulder
shoulder_df = data[shoulder][['FERC_time', 'PRICEINDOLPERMWH']]

**Test for summer**

In [55]:
summer_peak = summer_df['FERC_time'] == 'peak' #sample peak
summer_off_peak = summer_df['FERC_time'] == 'off_peak' #sample peak
summer_p_df = summer_df[summer_peak][['PRICEINDOLPERMWH']]
summer_op_df = summer_df[summer_off_peak][['PRICEINDOLPERMWH']]

summer_p_sample = len(summer_p_df) #sample size 
summer_op_sample = len(summer_op_df)
summer_p_mean = summer_p_df.mean() #sample mean
summer_op_mean = summer_op_df.mean()
summer_p_std = summer_p_df.std() #sample standard deviation
summer_op_std = summer_op_df.std()

print("summer peak")
print("mean:",summer_p_sample)
print("mean:",summer_p_mean)
print("std:",summer_p_std)
print("summer off-peak")
print("mean:",summer_op_sample)
print("mean:",summer_op_mean)
print("std:",summer_op_std)

summer peak
mean: 8283
mean: PRICEINDOLPERMWH    28.127083
dtype: float64
std: PRICEINDOLPERMWH    14.181533
dtype: float64
summer off-peak
mean: 7249
mean: PRICEINDOLPERMWH    23.698316
dtype: float64
std: PRICEINDOLPERMWH    8.26947
dtype: float64


In [56]:
#use z test since sample size > 30
#compute z value
summer_zvalue = (summer_p_mean - summer_op_mean) / (math.sqrt(((summer_p_std**2)/summer_p_sample) + ((summer_op_std**2)/summer_op_sample)))
summer_zvalue

PRICEINDOLPERMWH    24.119969
dtype: float64

In [57]:
#compute p-value
summer_pvalue = st.norm.sf(abs(summer_zvalue))*2
summer_pvalue

array([  1.54325166e-128])

**Test for winter**

In [58]:
winter_peak = winter_df['FERC_time'] == 'peak' #sample peak
winter_off_peak = winter_df['FERC_time'] == 'off_peak' #sample peak
winter_p_df = winter_df[winter_peak][['PRICEINDOLPERMWH']]
winter_op_df = winter_df[winter_off_peak][['PRICEINDOLPERMWH']]

winter_p_sample = len(winter_p_df) #sample size 
winter_op_sample = len(winter_op_df)
winter_p_mean = winter_p_df.mean() #sample mean
winter_op_mean = winter_op_df.mean()
winter_p_std = winter_p_df.std() #sample standard deviation
winter_op_std = winter_op_df.std()

print("winter peak")
print("mean:",winter_p_sample)
print("mean:",winter_p_mean)
print("std:",winter_p_std)
print("winter off-peak")
print("mean:",winter_op_sample)
print("mean:",winter_op_mean)
print("std:",winter_op_std)

winter peak
mean: 7951
mean: PRICEINDOLPERMWH    22.464182
dtype: float64
std: PRICEINDOLPERMWH    8.904606
dtype: float64
winter off-peak
mean: 5672
mean: PRICEINDOLPERMWH    21.122593
dtype: float64
std: PRICEINDOLPERMWH    8.102258
dtype: float64


In [59]:
#use z test since sample size > 30
#compute z value
winter_zvalue = (winter_p_mean - winter_op_mean) / (math.sqrt(((winter_p_std**2)/winter_p_sample) + ((winter_op_std**2)/winter_op_sample)))
winter_zvalue

PRICEINDOLPERMWH    9.139704
dtype: float64

In [60]:
#compute p-value
winter_pvalue = st.norm.sf(abs(winter_zvalue))*2
winter_pvalue

array([  6.26236070e-20])

**Test for shoulder**

In [61]:
shoulder_peak = shoulder_df['FERC_time'] == 'peak' #sample peak
shoulder_off_peak = shoulder_df['FERC_time'] == 'off_peak' #sample peak
shoulder_p_df = shoulder_df[shoulder_peak][['PRICEINDOLPERMWH']]
shoulder_op_df = shoulder_df[shoulder_off_peak][['PRICEINDOLPERMWH']]

shoulder_p_sample = len(shoulder_p_df) #sample size 
shoulder_op_sample = len(shoulder_op_df)
shoulder_p_mean = shoulder_p_df.mean() #sample mean
shoulder_op_mean = shoulder_op_df.mean()
shoulder_p_std = shoulder_p_df.std() #sample standard deviation
shoulder_op_std = shoulder_op_df.std()

print("shoulder peak")
print("mean:",shoulder_p_sample)
print("mean:",shoulder_p_mean)
print("std:",shoulder_p_std)
print("shoulder off-peak")
print("mean:",shoulder_op_sample)
print("mean:",shoulder_op_mean)
print("std:",shoulder_op_std)

shoulder peak
mean: 19982
mean: PRICEINDOLPERMWH    26.189026
dtype: float64
std: PRICEINDOLPERMWH    10.384733
dtype: float64
shoulder off-peak
mean: 13358
mean: PRICEINDOLPERMWH    22.241931
dtype: float64
std: PRICEINDOLPERMWH    7.312106
dtype: float64


In [62]:
#use z test since sample size > 30
#compute z value
shoulder_zvalue = (shoulder_p_mean - shoulder_op_mean) / (math.sqrt(((shoulder_p_std**2)/shoulder_p_sample) + ((shoulder_op_std**2)/shoulder_op_sample)))
shoulder_zvalue

PRICEINDOLPERMWH    40.712039
dtype: float64

In [63]:
#compute p-value
shoulder_pvalue = st.norm.sf(abs(shoulder_zvalue))*2
shoulder_pvalue

array([ 0.])

The above pvalue shows that the results are significant at the .01 level so we reject the null and accept the hypothesis that the prices between peridos are significant.

In order to measure how different we can compute the relative difference.

In [64]:
relative_difference_sum = (summer_p_mean-summer_op_mean)/summer_p_mean
relative_difference_win = (winter_p_mean-winter_op_mean)/winter_p_mean
relative_difference_sho = (shoulder_p_mean-shoulder_op_mean)/shoulder_p_mean

print("summer difference:",relative_difference_sum)
print("winter difference:",relative_difference_win)
print("shoulder difference:",relative_difference_sho)

summer difference: PRICEINDOLPERMWH    0.157456
dtype: float64
winter difference: PRICEINDOLPERMWH    0.059721
dtype: float64
shoulder difference: PRICEINDOLPERMWH    0.150716
dtype: float64


** Conclusion **

According to this analysis prices are different between the peak and off peak periods.

The difference is mot pronounced during the Summer and Shoulder periods at ~15% and less pronounced during the winter period at ~6%.